rightscale/rsc/client.go

```diff
@@ -722,3 +729,29 @@ func (inFields Fields) onlyPopulated() Fields {
 	}
 	return outFields
 }
+
+// analyzeSource checks that the defition of the source is valid, returning an error if it's not
+// If the definition is valid, expectsOuputs boolean indicates if the defition includes the "return" keyword,
+// which indicates that output values are expected.
+func analyzeSource(source string) (expectsOutputs bool, err error) {
+	const validDefinition = "^[[:blank:]]*define[[:blank:]]*[\\w_\\.]+[[:blank:]]*\\([@$\\w _,]*\\)[[:blank:]]+(return.*)?do"
+
+	r, _ := regexp.Compile(validDefinition)
```

Comment:
Use `MustCompile` instead of ignoring the error. In general ignoring errors is a big no no. I didn't comment on line 662 because we know the value was previously a JSON value. But even there you could consider handling the error.

Reply:
Probably doesn't change much in practice since it would be invalid syntax but to be more precise the regexp could include a blank character class after the `return`.